tests/relay/test_publish.py

```diff
@@ -1,34 +1,231 @@
+
+    def test_publish_with_missing_payload(self):
+        message = {"contentTopic": self.test_content_topic, "timestamp": int(time() * 1e9)}
+        try:
+            self.node1.send_message(message, self.test_pubsub_topic)
+            raise AssertionError("Publish with missing payload worked!!!")
+        except Exception as ex:
+            assert "Bad Request" in str(ex) or "Internal Server Error" in str(ex)
+
+    def test_publish_with_payload_less_than_one_mb(self):
+        payload_length = 1024 * 1023
+        logger.debug("Running test with payload length of %s bytes", payload_length)
+        message = self.create_message(payload=to_base64("a" * (payload_length)))
+        self.check_published_message_reaches_peer(message, message_propagation_delay=2)
+
+    def test_publish_with_payload_equal_or_more_than_one_mb(self):
+        payload_length = 1024 * 1023
+        for payload_length in [1024 * 1024, 1024 * 1024 * 10]:
+            logger.debug("Running test with payload length of %s bytes", payload_length)
+            message = self.create_message(payload=to_base64("a" * (payload_length)))
+            try:
+                self.check_published_message_reaches_peer(message, message_propagation_delay=2)
+                raise AssertionError("Duplicate message was retrieved twice")
+            except Exception as ex:
+                assert "Peer node couldn't find any messages" in str(ex)
+
+    def test_publish_with_valid_content_topics(self):
         failed_content_topics = []
         for content_topic in SAMPLE_INPUTS:
             logger.debug("Running test with content topic %s", content_topic["description"])
-            message = MessageRpcQuery(payload=to_base64(self.test_payload), contentTopic=content_topic["value"])
+            message = self.create_message(contentTopic=content_topic["value"])
             try:
                 self.check_published_message_reaches_peer(message)
             except Exception as e:
-                logger.error("ContentTopic %s failed: %s", {content_topic["description"]}, {str(e)})
+                logger.error("ContentTopic %s failed: %s", content_topic["description"], str(e))
                 failed_content_topics.append(content_topic)
         assert not failed_content_topics, f"ContentTopics failed: {failed_content_topics}"
+
+    def test_publish_with_invalid_content_topics(self):
+        success_content_topics = []
+        for content_topic in INVALID_CONTENT_TOPICS:
+            logger.debug("Running test with contetn topic %s", content_topic["description"])
+            message = self.create_message(contentTopic=content_topic["value"])
+            try:
+                self.node1.send_message(message, self.test_pubsub_topic)
+                success_content_topics.append(content_topic)
+            except Exception as ex:
+                assert "Bad Request" in str(ex) or "Internal Server Error" in str(ex)
```

Comment:
Those are negative scenarios / error handling tests.
It's expected for them to fail. The test will fail when the exception will not be triggered

Reply:
Right, but why check the exception string contains either `Bad Request` or `Internal Server Error`? 

Shouldn't all (even ones that don't contain those strings) exceptions make the test fail?

Reply:
So in this case, when passing an invalid value in the request payload I would expect Bad Request and only pass the test in that case. However there are inconcistencies for now between how nwaku and gowaku behaven and we get "Internal Server Error" in some cases (I hope that this will be fixed soon). 
However there can be other error messages like Not Found, Request Timeout etc that I want to catch and fail the test in that case.
There is a case when a request crashed the container. I want to catch such cases. That's why I check the message string and only pass the tests if the error is what I;m expecting

Reply:
> Right, got it!
> 
> Note for the future: Might be handy to map those errors to their own exceptions.

Agreed, when both nodes behave the same, now it would complicate the code